File: src/TinRoll.Test/Mapper/QuestionMapperTests.cs
//using System;
//using System.Collections.Generic;
//using System.Text;
//using TinRoll.Data.Entities;
//using TinRoll.Logic.Mapper;
//using TinRoll.Shared;
//using Xunit;

//namespace TinRoll.Test.Mapper
//{
//    public class QuestionMapperTests
//    {
//        [Fact]
//        public void Test_Question_ToQuestionDto()
//        {
//            var question = new Question
//            {
//                Id = 1,
//                Title = "Question Title",
//                Text = "Question Text",
//                CreatedDate = DateTime.UtcNow,
//                UpdatedDate = DateTime.UtcNow,
//                UserId = 1
//            };

//            var questionDto = QuestionMapper.ToDto(question);

//            Assert.NotNull(questionDto);
//            Assert.Equal(question.Id, questionDto.Id);
//            Assert.Equal(question.Title, questionDto.Title);
//            Assert.Equal(question.Text, questionDto.Text);
//            Assert.Equal(question.CreatedDate, questionDto.CreatedDate);
//            Assert.Equal(question.UpdatedDate, questionDto.UpdatedDate);
//            Assert.Equal(question.UserId, questionDto.UserId);
//        }

//        [Fact]
//        public void Test_QuestionDto_To_Question()
//        {
//            var questionDto = new QuestionDto
//            {
//                Id = 1,
//                Title = "Question Title",
//                Text = "Question Text",
//                CreatedDate = DateTime.UtcNow,
//                UpdatedDate = DateTime.UtcNow,
//                UserId = 1
//            };

//            var question = QuestionMapper.ToDb(questionDto);

//            Assert.NotNull(question);
//            Assert.Equal(questionDto.Id, question.Id);
//            Assert.Equal(questionDto.Title, question.Title);
//            Assert.Equal(questionDto.Text, question.Text);
//            Assert.Equal(questionDto.CreatedDate, question.CreatedDate);
//            Assert.Equal(questionDto.UpdatedDate, question.UpdatedDate);
//            Assert.Equal(questionDto.UserId, question.UserId);
//        }
//    }
//}

File: src/TinRoll.Test/Logic/Mapper/QuestionMapperTests.fs
module Logic.Mapper.QuestionMapperTests

open Xunit
open TinRoll.Data
open System
open TinRoll.Logic.Mapper
open TinRoll.Shared
open NFluent

[<Fact>]
let ``Test Question db to dto`` () =
    let mutable db = new Question(1, "Title", "Text", 23, DateTime.MinValue, DateTime.MinValue)
    db.User <- new User()
    db.User.Id <- 23
    let dto = QuestionMapper.ToDto(db)

    Check.That(dto.Id).Equals(db.Id) |> ignore
    Check.That(dto.Title).Equals(db.Title) |> ignore
    Check.That(dto.Text).Equals(db.Text) |> ignore
    Check.That(dto.UserId).Equals(db.UserId) |> ignore
    //Check.That(dto.User.Id).Equals(db.User.Id) |> ignore
    Check.That(dto.CreatedDate).Equals(db.CreatedDate) |> ignore
    Check.That(dto.UpdatedDate).Equals(db.UpdatedDate) |> ignore

[<Fact>]
let ``Question Dto to Db`` () =
    let mutable dto = new QuestionDto(Id = 1, Title = "Title", Text = "Text", UserId = 24, CreatedDate = DateTime.MinValue, UpdatedDate = DateTime.MinValue)
    dto.User <- new UserDto()
    dto.User.Id <- 24
    let db = QuestionMapper.ToDb(dto)

    Check.That(db.Id).Equals(dto.Id) |> ignore
    Check.That(db.Title).Equals(dto.Title) |> ignore
    Check.That(db.Text).Equals(dto.Text) |> ignore
    Check.That(db.UserId).Equals(dto.UserId) |> ignore
    Check.That(db.CreatedDate).Equals(dto.CreatedDate) |> ignore
    Check.That(db.UpdatedDate).Equals(dto.UpdatedDate) |> ignore

